### PR TITLE
Prometheus config updates

### PR DESCRIPTION
**Pull request by aweiteka**
Add a liveness probe to auto-reload config when hash changes
Increase retention to 3d matching production default

### CODE REVIEW

**Comment by aweiteka** [comment]
cc @jeremyeder

**Comment by aweiteka** [comment]
Related to upstream origin PR https://github.com/openshift/origin/pull/18391

File: roles/openshift_prometheus/defaults/main.yaml [hunk 1]
```diff
@@ -29,7 +29,7 @@ l_openshift_prometheus_serviceaccount_annotations:
 openshift_prometheus_additional_rules_file: null
 
 #prometheus application arguments
-openshift_prometheus_args: ['--storage.tsdb.retention=6h']
+openshift_prometheus_args: ['--storage.tsdb.retention=3d']
```
Comment (zgalor) [inline roles/openshift_prometheus/defaults/main.yaml]: I don't see that change in the template or in openshift/origin#18391. Is it in a different open PR in origin?

**Comment by jeremyeder** [comment]
@aweiteka there are two other changes.  Did you want to fold them in here?
```
20180202 12:03:11<jerms> storage.tsdb.retention bumped from 6h to 3d
20180202 12:03:25<jerms> storage.tsdb.max-block-duration bumped from 36m to 432m
20180202 12:03:38<jerms> storage.tsdb.min-block-duration bumped from 2m to 2h
```

**Comment by aweiteka** [comment]
> 20180202 12:03:11<jerms> storage.tsdb.retention bumped from 6h to 3d 
in this PR

> 20180202 12:03:25<jerms> storage.tsdb.max-block-duration bumped from 36m to 432m 
Per CLI help `--storage.tsdb.max-block-duration=<duration> Maximum duration compacted blocks may span. (Defaults to 10% of the retention period)`. With this PR retention is 3d. (24*3*60)*.1 = 432 so I think we're good.

> 20180202 12:03:38<jerms> storage.tsdb.min-block-duration bumped from 2m to 2h
2h is default

File: roles/openshift_prometheus/defaults/main.yaml (hunk 1 above)
Comment (aweiteka) [reply roles/openshift_prometheus/defaults/main.yaml]: We need a separate PR there.

**Comment by pgier** [comment]
/lgtm
I believe this will also fix this bz: https://bugzilla.redhat.com/show_bug.cgi?id=1538434

File: roles/openshift_prometheus/templates/prometheus.j2 [hunk 2]
```diff
@@ -82,6 +82,20 @@ spec:
...
+        livenessProbe:
+          exec:
+            command:
+            - /bin/bash
+            - -c
+            - |-
+              set -euo pipefail;
+              touch /tmp/prometheusconfig.hash;
+              if [[ $(find /etc/prometheus -type f | sort | xargs md5sum | md5sum) != $(cat /tmp/prometheusconfig.hash) ]]; then
+                find /etc/prometheus -type f | sort | xargs md5sum | md5sum > /tmp/prometheusconfig.hash;
```
Comment (pgier) [inline roles/openshift_prometheus/templates/prometheus.j2]: Could use a variable for the hash to avoid repeating the find/md5 commands

**Comment by davidaah** [comment]
@sdodson @pgier is it possible to cherry pick this to 3.7?

**Pull request merged**